app/components/reaction_picker/reaction_picker.js

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Image,
+    View,
+    TouchableWithoutFeedback,
+} from 'react-native';
+
+import Emoji from 'app/components/emoji';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {hapticFeedback} from 'app/utils/general';
+import addReactionIcon from 'assets/images/icons/reaction.png';
+import {REACTION_PICKER_HEIGHT} from '../../screens/post_options/post_options_utils';
+
+export default class Reaction extends PureComponent {
+    static propTypes = {
+        addReaction: PropTypes.func.isRequired,
+        openReactionScreen: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        recentEmojis: PropTypes.array,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedIndex: -1,
+            defaultEmojis: [
+                'thumbsup',
+                'smiley',
+                'white_check_mark',
+                'heart',
+                'eyes',
+                'raised_hands',
+            ],
+        };
+    }
+
+    handlePress = (emoji, index) => {
+        this.setState({
+            selectedIndex: index,
+        });
+        hapticFeedback();
+        setTimeout(() => {
+            this.props.addReaction(emoji);
+        }, 250);
+    }
+
+    render() {
+        const {
+            theme,
+        } = this.props;
+        const style = getStyleSheet(theme);
+        const list = [];
+
+        // Mixing recent emojis with default list and removing duplicates
+        const emojis = Array.from(new Set(this.props.recentEmojis.concat(this.state.defaultEmojis)));
+
+        emojis.forEach((emoji, index) => {
+            if (index < 6) {
+                const button = (
+                    <TouchableWithoutFeedback onPress={() => this.handlePress(emoji, index)}>
```

Comment:
I'm going to let it slide as this component is not heavy at all, but this is causing an unnecessary re-render when the `selectedIndex`  updates